src/fshtypes/ParamRuleSet.ts

```diff
@@ -0,0 +1,31 @@
+import escapeRegExp from 'lodash/escapeRegExp';
+import { FshEntity } from '.';
+
+export class ParamRuleSet extends FshEntity {
+  parameters: string[];
+  contents: string;
+
+  constructor(public name: string) {
+    super();
+    this.parameters = [];
+    this.contents = '';
+  }
+
+  applyParameters(values: string[]): string {
+    const paramUsage = new RegExp(`{(${this.parameters.map(escapeRegExp).join('|')})}`, 'g');
```

Comment:
How would you feel about allowing for optional whitespace between the `{`/`}` and parameter names?  E.g., right now, you support `{cheese}` but not `{ cheese }`.  I think it might be nice to allow for that whitespace (as I think I personally would use it).